### DISCUSSION TITLE
synthetic contirbutions

### DISCUSSION THREAD

**Discussion by younesStrittmatter** [AutoResearch/autora#594] [Module Contributions]
Should we use the following format for synthetic runners that are not pulled into autora-synthetic:
```
autora[experiment-runner-synthetic-...]
```
or
```
autora[synthetic-...]
```
?
The first one aligns with the import but is longer. For example, the import for the abstract equation package looks like this:
```
from autora.experiment_runner.synthetic.abstract.equation import equation_experiment
```

**Comment by younesStrittmatter** [comment]
Related:
we have optional dependencies for experiment-runners, should we add the synthetic ones as optional dependencies for the runners, or for a new category (synthetic) or for a subcategory of experiment-runners? Or are there other ideas?

**Comment by chadcwilliams** [comment]
I prefer `autora[synthetic-...]` however I think sticking to our formatting is the better idea here: `autora[experiment-runner-synthetic-...]` despite it leading to larger import statements. 

One way to make the import statements slightly shorter, is to allow for the dropping of the synthetic category on (e.g., abstract, psychology):
`from autora.experiment_runner.synthetic.abstract.equation import equation_experiment` 
becomes
`from autora.experiment_runner.synthetic import equation_experiment`

That's not to say that the categories do not exist, but rather you can import them using either method above. So, we'd need an `__init__` file  in the synthetic folder that accumulates all runners across all subfolders as a shortcut.

**Reply by younesStrittmatter** [reply]
not sure how the dropping of the category would work for optional dependencies though. I honestly think it wouldn't (or at least not straightforward)

**Reply by chadcwilliams** [reply]
I just mean for importing, not installing.

**Comment by younesStrittmatter** [comment]
Decision:
Full path in both the import and installation, make convenience optional dependencies:
- experiment-runners-all
- experiment-runners-synthetic-all
